runtime/src/read_only_accounts_cache.rs

```diff
@@ -71,21 +75,27 @@ impl ReadOnlyAccountsCache {
     }
 
     pub(crate) fn load(&self, pubkey: Pubkey, slot: Slot) -> Option<AccountSharedData> {
-        let key = (pubkey, slot);
-        let Some(mut entry) = self.cache.get_mut(&key) else {
-            self.misses.fetch_add(1, Ordering::Relaxed);
-            return None;
-        };
-        self.hits.fetch_add(1, Ordering::Relaxed);
-        // Move the entry to the end of the queue.
-        // self.queue is modified while holding a reference to the cache entry;
-        // so that another thread cannot write to the same key.
-        {
-            let mut queue = self.queue.lock().unwrap();
-            queue.remove(entry.index);
-            entry.index = queue.insert_last(key);
-        }
-        Some(entry.account.clone())
+        let (account, load_us) = measure_us!({
+            let key = (pubkey, slot);
+            let Some(mut entry) = self.cache.get_mut(&key) else {
+                self.misses.fetch_add(1, Ordering::Relaxed);
+                return None;
```

Comment:
shouldn't this only measure duration for cache-hits?
cache misses would have a very different distribution and I guess wouldn't be interesting.

Reply:
Yes, we could break out hits vs. misses. But, the write vs. read lock of the dash map are relevant for both hits and misses, for example.
Also, we hit 95% of the time, making the distinction between hit and miss not as relevant as it could be.
![image](https://github.com/solana-labs/solana/assets/75863576/c3a6146b-4dda-4897-bd4d-f77acbb8bcf2)

And, hits and misses both are part of the overall load time metrics from callers.
I imagined several possible metrics:
1. miss lookup time in read cache (assuming that we can't separate lock time vs lookup time if we're in a `DashMap`, for example)
2. hit lookup time in read cache
3. hit lock time in eviction queue
4. hit update eviction time
5. hit + miss total time, including miss and update eviction queue and acquiring lock times, cloning account, etc.
6. your suggestion: only hits, total time from lookup to eviction queue update and account clone.
I chose to do #5 to just keep things simple. Next time someone is investigating contributors to load time (which @brooks is doing now), this metric will be available to encompass ALL time we spent attempting to load anything from the read cache.

wdyt?